Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -17,11 +17,26 @@ class CustomerInfoResponseHandler {
 
     private let offlineCreator: OfflineCustomerInfoCreator?
     private let userID: String
+    private let failIfInvalidSubscriptionKeyDetectedInDebug: Bool
+    private let isDebug: () -> Bool
 
     /// - Parameter offlineCreator: can be `nil` if offline ``CustomerInfo`` shouldn't or can't be computed.
-    init(offlineCreator: OfflineCustomerInfoCreator?, userID: String) {
+    init(
+        offlineCreator: OfflineCustomerInfoCreator?,
+        userID: String,
+        failIfInvalidSubscriptionKeyDetectedInDebug: Bool,
+        isDebug: @escaping () -> Bool = {
```

Comment:
Added this in to make it possible to inject the debug value during unit tests

Reply:
I'm thinking we could even make this an auto-executable lambda:

```swift
isDebug: Bool = {
            #if DEBUG
            return true
            #else
            return false
            #endif
        }()
```

Reply:
I'm down with the idea of making it an autoExecutable lambda! Making it a variable that can be overridden later feels risky, in the unlikely scenario that it gets modified in an unintended way at runtime 🤷‍♂️

Will make the change!